### ISSUE TITLE
MGLShapeSource initWithIdentifier:shape:options causing unrecognized selector crash

### ISSUE THREAD

**Issue by esamelson** [mapbox/mapbox-gl-native#7467]
**Platform:** iOS
**Mapbox SDK version:** 3.4.0-beta.5

Hi,

I'm trying to create an `MGLShapeSource` from GeoJSON string data as outlined in the release notes for beta.5, but I'm getting an unexpected crash.

Here is some example GeoJSON data I've tried:

```
{"type":"FeatureCollection","features":[{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim0","hardwareUid":0,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.25230478341537,44.59403626724078]}},{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim1","hardwareUid":1,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.25342939980773,44.603257616057235]}},{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim2","hardwareUid":2,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.26737075331366,44.55712500986935]}},{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim3","hardwareUid":3,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.26125718501447,44.567769270591185]}},{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim4","hardwareUid":4,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.26125804853051,44.56776755616605]}},{"type":"Feature","properties":{"avlColor":"ffff8800","displayName":"Avl Sim5","hardwareUid":5,"timeStamp":null},"geometry":{"type":"Point","coordinates":[-123.26125718501447,44.567769270591185]}}]}
```

This is the code I'm using:
```
NSData *data = [geojsonDataString dataUsingEncoding:NSUTF8StringEncoding];
NSError *encodeError;
MGLShape *sourceShape = [MGLShape shapeWithData:data encoding:NSUTF8StringEncoding error:&encodeError];
MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:@"testtest" shape:sourceShape options:nil];
```

The third line seems to run fine and returns an `MGLShapeCollectionFeature` with all of the correct points. However, the fourth line throws the following error:

```
ExceptionsManager.js:71 Exception thrown while executing UI block: -[NSNull objCType]: unrecognized selector sent to instance
```

Weirdly, the code runs perfectly fine if I pass in a geoJSON string with no data, such as `{"type":"FeatureCollection","features":[]}`.

Wondering if this is related to #7456? But I'm only trying to deal with `MGLShape`s. Any thoughts or help would be much appreciated!

**Comment by 1ec5** [comment]
This isn’t related to #7456. I can reproduce the issue:

```
2016-12-17 15:32:38.913 xctest[41798:10476414] -[NSNull objCType]: unrecognized selector sent to instance 0x7fff79ba3780
```

```
#0	0x00007fff94b7df4e in objc_exception_throw ()
#1	0x00007fff9b9d410d in -[NSObject(NSObject) doesNotRecognizeSelector:] ()
#2	0x00007fff9b8da4d1 in ___forwarding___ ()
#3	0x00007fff9b8da058 in __forwarding_prep_0___ ()
#4	0x0000000104cab3fc in ::-[NSExpression(MGLAdditions) mgl_convertedValueWithValue:](id) at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/NSExpression+MGLAdditions.mm:60
#5	0x0000000104caaadc in ::-[NSExpression(MGLAdditions) mgl_filterValue]() at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/NSExpression+MGLAdditions.mm:21
#6	0x0000000104cac904 in ::-[NSDictionary(MGLAdditions) mgl_propertyMap]() at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/NSDictionary+MGLAdditions.mm:18
#7	0x0000000104c59114 in mbglFeature(mapbox::geometry::feature<double>, objc_object*, NSDictionary*) at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/MGLFeature.mm:335
#8	0x0000000104c58deb in ::-[MGLPointFeature geoJSONObject]() at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/MGLFeature.mm:37
#9	0x0000000104c5bddf in ::-[MGLShapeCollectionFeature geoJSONObject]() at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/MGLFeature.mm:178
#10	0x0000000104ce4651 in ::-[MGLShapeSource setShape:](MGLShape *) at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/MGLShapeSource.mm:102
#11	0x0000000104ce2ff4 in ::-[MGLShapeSource initWithIdentifier:shape:options:](NSString *, MGLShape *, NSDictionary *) at /Users/mxn/hub/mapbox-gl-native-2/platform/darwin/src/MGLShapeSource.mm:53
```

**Comment by 1ec5** [comment]
A fix is coming in #7472. Thanks for reporting this bug!

**Comment by 1ec5** [comment]
Fixed in #7472 on the release-ios-v3.4.0 branch.
